### PR TITLE
Replaced pyyaml with ruamel.yaml

### PR DESCRIPTION
**Pull request by UjjwalAyyangar**
* Created a git [mirror](https://github.com/UjjwalAyyangar/ruamel-yaml) for ruamel.yaml.
* Added ruamel.yaml as a submodule.
* Added support for installing ruamel.yaml from pip.

Addresses issue #282

### CODE REVIEW

**Comment by UjjwalAyyangar** [comment]
Encountered an installation issue with the build. I will fix it and notify when completed.

File: cli/setup.py [hunk 1]
```diff
@@ -19,7 +19,8 @@
     include_package_data=True,
     install_requires=[
         'click',
-        'pyyaml'
+        'pyyaml',
+        'ruamel.yaml<0.15'
```
Comment (ivotron) [inline cli/setup.py]: do we need to strictly use a version < 0.15? Also, I was under the impression that ruamel replaces pyyaml, why do we need to leave the two?

File: .gitmodules [hunk 2]
```diff
@@ -4,3 +4,6 @@
 [submodule "cli/bin/vendor/click"]
 	path = cli/bin/vendor/click
 	url = https://github.com/pallets/click/
+[submodule "cli/bin/vendor/ruamel-yaml"]
+	path = cli/bin/vendor/ruamel-yaml
+	url = https://github.com/UjjwalAyyangar/ruamel-yaml
```
Comment (ivotron) [inline .gitmodules]: we would also need to remove `pyyaml`

File: cli/Dockerfile [hunk 3]
```diff
@@ -1,9 +1,12 @@
 FROM python:2.7.14-alpine3.7
 
-RUN apk add --no-cache bash curl git sed grep bc coreutils
+RUN apk add --no-cache bash curl git sed grep bc coreutils && \
+    apk add --update g++ && \
+    rm /var/cache/apk/*
```
Comment (ivotron) [inline cli/Dockerfile]: can you please add `--no-cache` to the new `apk` line?

**Comment by UjjwalAyyangar** [comment]
@ivotron Completed :+1:

**Comment by ivotron** [comment]
Thanks a lot!

**Pull request merged**